webapp/channels/src/components/header_footer_template_route.tsx

```diff
@@ -2,33 +2,33 @@
 // See LICENSE.txt for license information.
 
 import React from 'react';
-import {Route} from 'react-router-dom';
+import {Route, RouteProps} from 'react-router-dom';
 
 const HeaderFooterTemplate = React.lazy(() => import('components/header_footer_template'));
 const LoggedIn = React.lazy(() => import('components/logged_in'));
 
-export const HFTRoute = ({component: Component, ...rest}) => (
+export const HFTRoute = ({component: Component, ...rest}: RouteProps) => (
```

Comment:
Maybe we should extend this to something like
```
interface Props extends RouteProps {
    component: React.ComponentType<RouteComponentProps>;
}
```
The reason being:
- We never render this component without component props
- We want the component prop to be always be passed as props

Similarly for below

Reply:
@M-ZubairAhmed please can you review the changes you requested in my new commit and let me know if it's okay.

Reply:
looks great thank you